tap_pendo/streams.py

```diff
@@ -378,8 +377,16 @@ def sync_substream(self, state, parent, sub_stream, parent_response):
 
                 # Filtering the visitors based on last updated to improve performance of visitor_history replication
                 if isinstance(parent, Visitors):
+                    if record['metadata'].get('pendo', {'donotprocess': False}).get('donotprocess'):
+                        # If any visitor is set the Do Not Process flag then Pendo will stop collecting events
+                        # from that visitor and will stop displaying guides to that visitor
+                        LOGGER.info("Record marked as 'Do Not Process': %s", record[parent.key_properties[0]])
+                        continue
+
                     last_processed = self.get_last_processed(state, sub_stream)
-                    parent_last_updated = datetime.fromtimestamp(float(record['metadata']['auto'][self.replication_key]) / 1000.0, timezone.utc)
+                    # If there is no last_update key then set 'last_updated=now()' and fetch the records from recent bookmark
+                    parent_last_updated = datetime.fromtimestamp(float(
+                        record['metadata']['auto'][self.replication_key]) / 1000.0, timezone.utc) if record['metadata']['auto'].get(self.replication_key) else now()
```

Comment:
140 characters on one line which is too long. Can you please align it properly. 100 chars per line are preferred.

Reply:
Done

Reply:
Do we know under what scenarios a record will not have a last_update key? Is it linked exclusively to visitors with do_not_process flags set?